### ISSUE TITLE
Partial refund does not update order post

### ISSUE THREAD

**Issue by prokopsdk** [woocommerce/woocommerce#28969]
I am not seeing a change on the post_modified field of the wp_posts table, when a partial refund is done.
I am seeing a new post created. Is this intended behaviour? I would expect the order post get updated upon a partial refund.

**Comment by barryhughes** [comment]
Thank you for the report and apologies for the delay in responding. We agree this is a probable bug/oversight.

**Comment by blogtutor** [comment]
I've recently run into a related issue.  When editing the products in a (pending) Order or existing Subscription, and saving that info by clicking "Save" in the metabox (not the main "Update" button), the `post_modified` field is not being updated.

Since those are actual changes to the order or subscription, WooCommerce should update the modified date accordingly.

(I discovered this because I use Metorik, a reporting tool that syncs data from WooCommerce.  They use the modified date to know when things have been updated, so they can keep their data in sync. I recently had an issue where a customer removed one (of two) products from an existing subscription... since the modified date didn't get changed, Metorik didn't sync and I didn't know about it.)

**Comment by barryhughes** [comment]
Thanks; yes I see the same. In my own testing it seems this can be mitigated by using the order editor screen `Update` button, *after* altering the line items, but I agree that should not be necessary (or else, the line item changes should not be persisted until the user clicks on `Update`—but that *probably* is not the path we will take, here).

**Comment by blogtutor** [comment]
> this can be mitigated by using the order editor screen `Update` button, _after_ altering the line items

Yep -- however, this workaround isn't available if a customer edits their order from their "My Account" area, (For example, if a customer [removes one subscription product from an existing subscription](https://woocommerce.com/document/subscriptions/customers-view/#section-5) that has [had] two products.)

I do think it's an odd UI/UX quirk that changes are saved inside the meta box automatically, without actually clicking "Update" on the entire order/subscription since this isn't standard for anything else inside the WordPress editor.  But since that probably isn't going to change anytime soon, I agree it would be a good (and relatively simple) fix to have any of these changes simply update the post_modified date.  Thanks! 😊

**Comment by barryhughes** [comment]
That's a good note, thank you.

**Comment by barryhughes** [comment]
✍🏽  Has potential implications for HPOS/sync.

**Comment by bryceadams** [comment]
Just another +1 on @blogtutor's suggestion. We see this a lot with other stores using Metorik where they can't easily see if an update was made on the order due to the `post_modified` field not changing. It would be really helpful for us + our users, and perhaps other plugins/apps relying on the `post_modified` field, to have it updated when this change happens. Thank you for any consideration 😄

**Comment by coreymckrill** [comment]
Some notes:

For the first issue, with the order modified date not changing when issuing a refund, this happens regardless of the status of HPOS. If HPOS is enabled and the WC Order Tables are set to be the primary data store, the `date_updated_gmt` field still does not change when adding a refund.

However, for the second issue, with changing line items in a pending order, if HPOS is enabled and the WC Order Tables are set to primary, the modified date _does_ change (including in the posts table `post_modified` field if syncing between the two tables is enabled). I would expect this because of [this line](https://github.com/woocommerce/woocommerce/blob/756fe32c048a6c8e4e81d33147398340e93adc84/plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php#L2120). My guess is that any order metadata change would update the modified date of the order when HPOS is being used. Which I guess means that there is no metadata associated with a refund, only a separate order_refund object.

So, it might be worth splitting this into two separate issues, or at least solving this issue with two separate PRs.

**Comment by coreymckrill** [comment]
Another note regarding the second issue: with HPOS enabled, modifying or deleting an order item updates the modified date, _but adding a new order item does not_. Without HPOS (i.e. posts table authoritative), none of those actions updates the modified date.

**Comment by coreymckrill** [comment]
Does the line item action update the order's modified date? ✅ Yes / ❌ No

| Action | CPT | HPOS |
|--------|--------|--------|
| Add product | ❌ | ❌ |
| Change product | ❌ | ✅ |
| Delete product | ❌ | ✅ |
| Add shipping | ❌ | ❌ |
| Change shipping | ❌ | ✅ | 
| Delete shipping | ❌ | ✅ | 
| Add fee | ❌ | ✅ | 
| Change fee | ❌ | ✅ | 
| Delete fee | ❌ | ✅ |

**Comment by zoupkat** [comment]
Hello there @nerrad ! <br>Pinging you since I see that Corey and Bryce are no longer around.<br>Was wondering about the progress of this issue (is there any ETA at this time?).<br>And can you please confirm if improvements happen on this, would it address the current expected behaviour "If the full order value is not refunded, the order status will not change to `Refunded` and the email that is sent to the shopper will refer to a “partial refund.”" that is mentioned in Step 7 here: [https://woocommerce.com/document/woocommerce-refunds/#automatic-refunds](<https://woocommerce.com/document/woocommerce-refunds/#automatic-refunds>) ?<br>Should I be creating a new Linear issue to ask for a Partial Refund order status to be added at some point and to automatically update to that when a partial refund is issued?<br>Can you please clarify what is possible and if I should be opening a new issue?<br>Thank you in advance!

**Comment by nerrad** [comment]
AFAIK there was no progress. I think a first step (given the age of the issue) would be to see if it's still reproducible and test expectations.

**Comment by flakas** [comment]
Hi @prokopsdk - thanks for reporting this, and for the detail you gave.

I know this has been open for a very long time, and I appreciate your patience. We're currently going through some older WooCommerce issues, identifying what still needs fixing, and tidying up what's already fixed or out of scope.

It looks like this has already been fixed in #37047, I could no longer reproduce it on WC 10.9.4.

I'm going to close this one. If you're still facing this issue on a current version, please reply with steps to reproduce and we'll happily take another look.
